x-pack/plugins/ml/public/application/datavisualizer/stats_datagrid/stats_datagrid.tsx

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo, useState } from 'react';
+
+import {
+  CENTER_ALIGNMENT,
+  EuiButtonIcon,
+  EuiFlexItem,
+  EuiIcon,
+  EuiInMemoryTable,
+  EuiText,
+  HorizontalAlignment,
+  LEFT_ALIGNMENT,
+  RIGHT_ALIGNMENT,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { EuiTableComputedColumnType } from '@elastic/eui/src/components/basic_table/table_types';
+import { FieldTypeIcon } from '../../components/field_type_icon';
+import { FieldVisConfig } from '../index_based/common';
+import { DataVisualizerFieldExpandedRow } from './expanded_row';
+import { DocumentStat } from './components/field_data_row/document_stats';
+import { DistinctValues } from './components/field_data_row/distinct_values';
+import { NumberContentPreview } from './components/field_data_row/number_content_preview';
+import { DataVisualizerIndexBasedAppState } from '../../../../common/types/ml_url_generator';
+import { useTableSettings } from '../../data_frame_analytics/pages/analytics_management/components/analytics_list/use_table_settings';
+import { TopValuesPreview } from './components/field_data_row/top_values_preview';
+import type { MlJobFieldType } from '../../../../common/types/field_types';
+const FIELD_NAME = 'fieldName';
+
+interface DataVisualizerDataGrid {
+  items: FieldVisConfig[];
+  pageState: DataVisualizerIndexBasedAppState;
+  updatePageState: (update: Partial<DataVisualizerIndexBasedAppState>) => void;
+}
+
+export type ItemIdToExpandedRowMap = Record<string, JSX.Element>;
+
+function getItemIdToExpandedRowMap(
+  itemIds: string[],
+  items: FieldVisConfig[]
+): ItemIdToExpandedRowMap {
+  return itemIds.reduce((m: ItemIdToExpandedRowMap, fieldName: string) => {
+    const item = items.find((fieldVisConfig) => fieldVisConfig[FIELD_NAME] === fieldName);
+    if (item !== undefined) {
+      m[fieldName] = <DataVisualizerFieldExpandedRow item={item} />;
+    }
+    return m;
+  }, {} as ItemIdToExpandedRowMap);
+}
+
+export const DataVisualizerDataGrid = ({
+  items,
+  pageState,
+  updatePageState,
+}: DataVisualizerDataGrid) => {
+  const [expandedRowItemIds, setExpandedRowItemIds] = useState<string[]>([]);
+  const [expandAll, toggleExpandAll] = useState<boolean | undefined>();
```

Comment:
can it be just `boolean`?

Reply:
Updated here https://github.com/elastic/kibana/pull/85726/commits/4ea6d174728033f29d81392da47db1f3b409d4a5